src/actions/index.js

```diff
@@ -0,0 +1,8 @@
+export const CREATE_BOOK = ({ book }, state) => [
+  ...state,
+  book,
+];
+
+export const REMOVE_BOOK = ({ book }, state) => state.filter(
+  ({ bookId }) => bookId !== book.bookId,
+);
```

Comment:
**[Required]** These are the actions that need to be dispatched. Passed arguments don't match with what you are getting in the reducer. For example, in the CREATE_BOOK, you are returning an array but in the reducer, you are expecting a type and book as arguments. Keep it simple, just return an object with type and payload.